spec/meilisearch/client/keys_spec.rb

```diff
@@ -1,60 +1,143 @@
 # frozen_string_literal: true
 
 RSpec.describe 'MeiliSearch::Client - Keys' do
-  it 'gets the list of keys' do
-    response = client.keys
-    expect(response).to be_a(Hash)
-    expect(response.count).to eq(2)
-    expect(response.keys).to contain_exactly('private', 'public')
-    expect(response['private']).to be_a(String)
-    expect(response['public']).to be_a(String)
-  end
+  context 'Test the default key roles' do
+    let(:public_key) { client.keys['results'].filter { |k| k['description'].start_with? 'Default Search API Key' }.first }
+    let(:private_key) { client.keys['results'].filter { |k| k['description'].start_with? 'Default Admin API Key' }.first }
 
-  it 'fails to get settings if public key used' do
-    public_key = client.keys['public']
-    new_client = MeiliSearch::Client.new(URL, public_key)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
-  end
+    it 'fails to get settings if public key used' do
+      new_client = MeiliSearch::Client.new(URL, public_key['key'])
+      expect do
+        new_client.index(random_uid).settings
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to get keys if private key used' do
-    private_key = client.keys['private']
-    new_client = MeiliSearch::Client.new(URL, private_key)
-    expect do
-      new_client.keys
-    end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
-  end
+    it 'fails to get keys if private key used' do
+      new_client = MeiliSearch::Client.new(URL, private_key['key'])
+      expect do
+        new_client.keys
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to search if no key used' do
-    new_client = MeiliSearch::Client.new(URL)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
-  end
+    it 'fails to search if no key used' do
+      new_client = MeiliSearch::Client.new(URL)
+      expect do
+        new_client.index(random_uid).settings
+      end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
+    end
 
-  it 'succeeds to search when using public key' do
-    uid = random_uid
-    public_key = client.keys['public']
-    index = client.create_index(uid)
-    response = index.add_documents(title: 'Test')
-    index.wait_for_pending_update(response['updateId'])
+    it 'succeeds to search when using public key' do
+      uid = random_uid
+      index = client.index(uid)
+      index.add_documents!(title: 'Test')
 
-    new_client = MeiliSearch::Client.new(URL, public_key)
-    response = new_client.index(uid).search('test')
-    expect(response).to have_key('hits')
-  end
+      new_client = MeiliSearch::Client.new(URL, public_key['key'])
+      response = new_client.index(uid).search('test')
```

Comment:
If we just make the "search" without adding documents we could make an assertion by just ensuring there were no meilisearch_api_errors. That way we could simplify a bit this spec 🦘

Reply:
We need to create an index to perform the search, so I can replace `add_documents!` with `create_index!`, but it does not simplify the spec :(

Reply:
I was thinking about something like this:

```rb
let(:public_client) { MeiliSearch::Client.new(URL, public_key['key']) }

it 'succeeds to search when using public key' do
  uid = random_uid

  expect {
    public_client.index(uid).search('test')
  }.not_to raise_meilisearch_api_error_with(...)
end
```

- public_client could be reused
- we only check if no errors are produced since we don’t care about the search result

(btw, I did this here on the comment, probably there are some typo)

Feel free to disagree as well haha 🍔

Reply:
If the index does not exist, `public_client.index(uid).search('test')` will return a 404 since the index does not exist. You cannot search in a non-existing index